org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/AbstractLanguageServerTest.xtend

```diff
@@ -726,6 +727,23 @@ abstract class AbstractLanguageServerTest implements Endpoint {
 		val result = new Document(1, fileInfo.contents).applyChanges(<TextEdit>newArrayList(changes.get()).reverse)
 		assertEqualsStricter(configuration.expectedText, result.contents)
 	}
+	
+	/**
+	 * @since 2.25
+	 */
+	protected def testFolding((FoldingConfiguration)=>void configurator) {
+		val extension configuration = new FoldingConfiguration
+		configuration.filePath = 'MyModel.' + fileExtension
+		configurator.apply(configuration)
+		
+		val fileInfo = initializeContext(configuration)
+		
+		val foldings = languageServer.foldingRange(new FoldingRangeRequestParams => [
+			textDocument = new TextDocumentIdentifier(fileInfo.uri)
+		]).get()
+		
+		assertEqualsStricter(configuration.expectedFoldings, String.join("", foldings.map[toString]));
```

Comment:
Instead of using `String.join("", foldings.map[toString])`, I would prefer using `foldings.toExpectation` and adding the following dispatch method:
```Xtend
protected dispatch def String toExpectation(FoldingRange foldingRange) {
	foldingRange.toString	
}
```
In that case, the code will remain consistent to the already existing actual value calculations.

Let's see what the other colleagues think about this suggestion.